cpp/tests/io/arrow_io_source_test.cpp

```diff
@@ -97,7 +97,7 @@ TEST_F(ArrowIOTest, S3FileSystem)
 
     close_s3_func = reinterpret_cast<decltype(close_s3_func)>(
       dlsym(whole_app, "_ZN5arrow2fs17EnsureS3FinalizedEv"));
-    if (close_s3_func) { CUDF_EXPECTS(close_s3_func().ok(), "Failed to finalize s3 filesystem"); }
+    if (close_s3_func) { ASSERT_TRUE(close_s3_func().ok()); }
```

Comment:
Seems this one should be `EXPECT_TRUE` so as to not prevent the call to `dlclose()`.